ci-operator/step-registry/cucushift/installer/check/azure/cpms/cucushift-installer-check-azure-cpms-commands.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# set the parameters we'll need as env vars
+AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
+AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
+AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
+
+# log in with az
+if [[ "${CLUSTER_TYPE}" == "azuremag" ]]; then
+    az cloud set --name AzureUSGovernment
+elif [[ "${CLUSTER_TYPE}" == "azurestack" ]]; then
+    if [ ! -f "${CLUSTER_PROFILE_DIR}/cloud_name" ]; then
+        echo "Unable to get specific ASH cloud name!"
+        exit 1
+    fi
+    cloud_name=$(< "${CLUSTER_PROFILE_DIR}/cloud_name")
+
+    AZURESTACK_ENDPOINT=$(cat "${SHARED_DIR}"/AZURESTACK_ENDPOINT)
+    SUFFIX_ENDPOINT=$(cat "${SHARED_DIR}"/SUFFIX_ENDPOINT)
+
+    if [[ -f "${CLUSTER_PROFILE_DIR}/ca.pem" ]]; then
+        cp "${CLUSTER_PROFILE_DIR}/ca.pem" /tmp/ca.pem
+        cat /usr/lib64/az/lib/python*/site-packages/certifi/cacert.pem >> /tmp/ca.pem
+        export REQUESTS_CA_BUNDLE=/tmp/ca.pem
+    fi
+    az cloud register \
+        -n ${cloud_name} \
+        --endpoint-resource-manager "${AZURESTACK_ENDPOINT}" \
+        --suffix-storage-endpoint "${SUFFIX_ENDPOINT}"
+    az cloud set --name ${cloud_name}
+    az cloud update --profile 2019-03-01-hybrid
+else
+    az cloud set --name AzureCloud
+fi
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+
+ocp_minor_version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f2)
+if (( ${ocp_minor_version} < 14 )); then
+    echo "CPMS failureDomain check is only available on 4.14+ cluster, skip the check!"
+    exit 0
+fi
+
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+INFRA_ID=$(jq -r .infraID ${SHARED_DIR}/metadata.json)
+RESOURCE_GROUP=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.resourceGroupName')
+if [[ -z "${RESOURCE_GROUP}" ]]; then
+    RESOURCE_GROUP="${INFRA_ID}-rg"
+fi
+REGION=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.region')
+
+readarray -t zones_setting_from_config < <(yq-go r ${INSTALL_CONFIG} 'controlPlane.platform.azure.zones[*]')
+if (( ${#zones_setting_from_config[@]} > 1 )); then
+    echo "zones_setting_from_config: ${zones_setting_from_config[*]}"
+    echo "Field zones contains more than 1 zone in install-config, no need to check CPMS failureDomain, skip the check!"
```

Comment:
For those cases that `no need to check CPMS failureDomain`, will the `failureDomains` field configured with something like "platform: Azure"? could we also make checking on that, If so, this checking will be a more complete test.

Reply:
yes, it's "platform: Azure" for the zone setting or instance types supporting more than 1 zone. 
good point, let me update to check more.